core/agents/orchestrator.py

```diff
@@ -226,12 +227,23 @@ def create_agent(self, prev_response: Optional[AgentResponse]) -> BaseAgent:
             return self.create_agent_for_step(state.current_step)
 
         if state.unfinished_iterations:
-            if state.current_iteration["description"]:
-                # Break down the next iteration into steps
+            if state.current_iteration["status"] == IterationStatus.HUNTING_FOR_BUG:
+                # Ask the Logger to check if more logs in the code are needed
+                return BugHunter(self.state_manager, self.ui)
+            elif (state.current_iteration["status"] == IterationStatus.AWAITING_LOGGING or
+                  state.current_iteration["status"] == IterationStatus.AWAITING_BUG_FIX or
+                  state.current_iteration["status"] == IterationStatus.IMPLEMENT_SOLUTION):
+                # Ask the Logger to ask user to test new logs
                 return Developer(self.state_manager, self.ui)
-            else:
-                # We need to iterate over the current task but there's no solution, as Pythagora
-                # is stuck in a loop, and ProblemSolver needs to find alternative solutions.
+            elif (state.current_iteration["status"] == IterationStatus.AWAITING_USER_TEST or
```

Comment:
same as above, dont reuse comment `# Ask the Logger to ask user to test new logs`

Reply:
done